app/services/trainees/create_from_hesa.rb

```diff
@@ -73,6 +72,11 @@ def trainee_record_source
     end
 
     def trn
+      # From the HESA data it appears that a TRN might not always be submitted with
+      # an update, even though a TRN has previously been submitted. This prevents a
+      # blank TRN in the HESA update from overwriting an existing TRN in register.
+      return trainee.trn if trainee.trn.present?
```

Comment:
ahhhhhh! nice catch. I've just seen a couple of trainees in trn_received with no trn - so maybe this happened to them.

Reply:
No, the `compact` statement would have removed all keys will `nil` values. I've had to this here because `compact` was originally brought in to stop existing `trn` from been nullifed, but we now need to be able to nullify withdrawal and deferral values.